### PR TITLE
[BCMSAI] Update BCM SAI debian package to 4.2.1.3 (6.5.19 hsdk)

### PR DESCRIPTION
**Pull request by smaheshm**
**- Why I did it**

Update BCM SAI debian package to 4.2.1.3 (6.5.19 hsdk)

**- How I did it**

Build debain package with BCMSAI 4.2.1.3 and 6.5.19 hsdk.

**- How to verify it**

Manual tests
```
 

admin@str-s6000-acs-8:~$ show version

SONiC Software Version: SONiC.HEAD.4898-c0167da34
Distribution: Debian 10.5
Kernel: 4.19.0-9-2-amd64
Build commit: c0167da34
Build date: Fri Sep 18 00:06:12 UTC 2020
Built by: johnar@jenkins-worker-2

Platform: x86_64-dell_s6000_s1220-r0
HwSKU: Force10-S6000
ASIC: broadcom
Serial Number: 1QBRX42
Uptime: 18:25:22 up 11 min,  1 user,  load average: 0.85, 2.01, 1.68

Docker images:
REPOSITORY                    TAG                   IMAGE ID            SIZE
docker-teamd                  HEAD.4898-c0167da34   7f98a0fc1bd4        391MB
docker-teamd                  latest                7f98a0fc1bd4        391MB
docker-database               HEAD.4898-c0167da34   86f7343195cc        360MB
docker-database               latest                86f7343195cc        360MB
docker-syncd-brcm             HEAD.4898-c0167da34   0469aa691b78        447MB
docker-syncd-brcm             latest                0469aa691b78        447MB
docker-router-advertiser      HEAD.4898-c0167da34   86e3b2a3f00a        359MB
docker-router-advertiser      latest                86e3b2a3f00a        359MB
docker-platform-monitor       HEAD.4898-c0167da34   cc2c8b2712d2        433MB
docker-platform-monitor       latest                cc2c8b2712d2        433MB
docker-lldp                   HEAD.4898-c0167da34   66f99a05fd09        387MB
docker-lldp                   latest                66f99a05fd09        387MB
docker-dhcp-relay             HEAD.4898-c0167da34   14979441d337        366MB
docker-dhcp-relay             latest                14979441d337        366MB
docker-orchagent              HEAD.4898-c0167da34   043fb9e5daa6        404MB
docker-orchagent              latest                043fb9e5daa6        404MB
docker-nat                    HEAD.4898-c0167da34   69677d0744df        393MB
docker-nat                    latest                69677d0744df        393MB
docker-fpm-frr                HEAD.4898-c0167da34   1ba9acdd051e        406MB
docker-fpm-frr                latest                1ba9acdd051e        406MB
docker-sflow                  HEAD.4898-c0167da34   89342a4153eb        394MB
docker-sflow                  latest                89342a4153eb        394MB
docker-snmp                   HEAD.4898-c0167da34   462bfd5b2e75        399MB
docker-snmp                   latest                462bfd5b2e75        399MB
docker-sonic-mgmt-framework   HEAD.4898-c0167da34   fb4517dd49c8        487MB
docker-sonic-mgmt-framework   latest                fb4517dd49c8        487MB
docker-sonic-telemetry        HEAD.4898-c0167da34   0d7aba7283ca        430MB
docker-sonic-telemetry        latest                0d7aba7283ca        430MB

admin@str-s6000-acs-8:~$ sudo monit status | grep ^Process
Process 'rsyslog'
Process 'telemetry'
Process 'dialout_client'
Process 'syncd'
Process 'dsserve'
Process 'orchagent'
Process 'portsyncd'
Process 'neighsyncd'
Process 'vrfmgrd'
Process 'vlanmgrd'
Process 'intfmgrd'
Process 'portmgrd'
Process 'buffermgrd'
Process 'nbrmgrd'
Process 'vxlanmgrd'
Process 'snmpd'
Process 'snmp_subagent'
Process 'sflowmgrd'
Process 'lldpd_monitor'
Process 'lldp_syncd'
Process 'lldpmgrd'
Process 'redis_server'
Process 'zebra'
Process 'fpmsyncd'
Process 'bgpd'
Process 'staticd'
Process 'bgpcfgd'


admin@str-s6000-acs-8:~$ show ip bgp summ

IPv4 Unicast Summary:
BGP router identifier 10.1.0.32, local AS number 65100 vrf-id 0
BGP table version 12909
RIB entries 13011, using 2394024 bytes of memory
Peers 24, using 502080 KiB of memory
Peer groups 4, using 256 bytes of memory


Neighbhor      V     AS    MsgRcvd    MsgSent    TblVer    InQ    OutQ  Up/Down      State/PfxRcd  NeighborName
-----------  ---  -----  ---------  ---------  --------  -----  ------  ---------  --------------  --------------
10.0.0.1       4  65200       3354       3389         0      0       0  00:07:24             6402  ARISTA01T2
10.0.0.5       4  65200       3387       3428         0      0       0  00:09:05             6402  ARISTA03T2
10.0.0.9       4  65200       3388       3429         0      0       0  00:09:06             6402  ARISTA05T2
10.0.0.13      4  65200       3388       3432         0      0       0  00:09:06             6402  ARISTA07T2
10.0.0.17      4  65200       3388       3429         0      0       0  00:09:06             6402  ARISTA09T2
10.0.0.21      4  65200       3387       3428         0      0       0  00:09:05             6402  ARISTA11T2
10.0.0.25      4  65200       3376       3412         0      0       0  00:08:31             6402  ARISTA13T2
10.0.0.29      4  65200       3388       3429         0      0       0  00:09:06             6402  ARISTA15T2
10.0.0.33      4  64001        145       3377         0      0       0  00:06:55                7  ARISTA01T0
10.0.0.35      4  64002        148       3382         0      0       0  00:07:08                6  ARISTA02T0
10.0.0.37      4  64003        166       3400         0      0       0  00:07:59                7  ARISTA03T0
10.0.0.39      4  64004        168       3403         0      0       0  00:08:07                6  ARISTA04T0
10.0.0.41      4  64005        164       3399         0      0       0  00:07:55                6  ARISTA05T0
10.0.0.43      4  64006        161       3397         0      0       0  00:07:48                6  ARISTA06T0
10.0.0.45      4  64007        174       3410         0      0       0  00:08:27                6  ARISTA07T0
10.0.0.47      4  64008        156       3392         0      0       0  00:07:33                6  ARISTA08T0
10.0.0.49      4  64009        171       3406         0      0       0  00:08:17                6  ARISTA09T0
10.0.0.51      4  64010        188       4096         0      0       0  00:09:09                6  ARISTA10T0
10.0.0.53      4  64011        155       3390         0      0       0  00:07:28                6  ARISTA11T0
10.0.0.55      4  64012        151       3385         0      0       0  00:07:16                6  ARISTA12T0
10.0.0.57      4  64013        174       3409         0      0       0  00:08:24                6  ARISTA13T0
10.0.0.59      4  64014        182       3418         0      0       0  00:08:50                6  ARISTA14T0
10.0.0.61      4  64015        163       3398         0      0       0  00:07:52                6  ARISTA15T0
10.0.0.63      4  64016        188       4096         0      0       0  00:09:08                6  ARISTA16T0

Total number of neighbors 24
admin@str-s6000-acs-8:~$ ping 10.0.0.1
PING 10.0.0.1 (10.0.0.1) 56(84) bytes of data.
64 bytes from 10.0.0.1: icmp_seq=1 ttl=64 time=7.81 ms


admin@str-s6000-acs-8:~$ show lldp neighbors  | grep SysName
    SysName:      str-22ex-235ab08
    SysName:      ARISTA07T2
    SysName:      ARISTA07T2
    SysName:      ARISTA09T2
    SysName:      ARISTA09T2
    SysName:      ARISTA11T2
    SysName:      ARISTA11T2
    SysName:      ARISTA01T2
    SysName:      ARISTA01T2
    SysName:      ARISTA03T2
    SysName:      ARISTA03T2
    SysName:      ARISTA05T2
    SysName:      ARISTA05T2
    SysName:      ARISTA13T2
    SysName:      ARISTA13T2
    SysName:      ARISTA15T2
    SysName:      ARISTA15T2
    SysName:      ARISTA01T0
    SysName:      ARISTA02T0
    SysName:      ARISTA04T0
    SysName:      ARISTA03T0
    SysName:      ARISTA12T0
    SysName:      ARISTA11T0
    SysName:      ARISTA13T0
    SysName:      ARISTA14T0
    SysName:      ARISTA15T0
    SysName:      ARISTA16T0
    SysName:      ARISTA05T0
    SysName:      ARISTA06T0
    SysName:      ARISTA08T0
    SysName:      ARISTA07T0
    SysName:      ARISTA09T0
    SysName:      ARISTA10T0


    admin@str-s6000-acs-8:~$ sudo grep -i crash /var/log/syslog.1
    Sep 18 18:13:12.303187 str-s6000-acs-8 INFO systemd[1]: Stopping Kernel crash dump capture service...
    Sep 18 18:14:22 sonic systemd[1]: Starting Kernel crash dump capture service...
    Sep 18 18:14:22 sonic systemd[1]: Starting Kernel crash dump capture service...
    Sep 18 18:14:25 sonic kdump-tools[349]: Starting kdump-tools: no crashkernel= parameter in the kernel cmdline ... failed!
    Sep 18 18:14:25 sonic kdump-tools[349]: Starting kdump-tools: no crashkernel= parameter in the kernel cmdline ... failed!
    Sep 18 18:14:27 sonic systemd[1]: Started Kernel crash dump capture service.
    Sep 18 18:14:27 sonic systemd[1]: Started Kernel crash dump capture service.
    Sep 18 18:16:29.893901 str-s6000-acs-8 WARNING syncd#syncd: :- discover: skipping since it causes crash: SAI_STP_ATTR_BRIDGE_ID
    admin@str-s6000-acs-8:~$ sudo grep -i crash /var/log/syslog
    admin@str-s6000-acs-8:~$ 
```

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

BRCM_SAI_VER "4.2.1.3"
OCP_SAI_VER  "1.6.5"
BCM SAI changelog (git log --pretty=oneline REL_GA1_3.7.5.1...REL_GA_4.2.1.3)
```
aea987188f7d4942fceef4b0abbe2afbce7f791d Add kbp file
6f3e31a5b05d59ddfce6ee01eceb38f059789d3b Remove old jink files.
3f53c17415b946e67cf253417a8739894d0ba489 4.2 GA release
06dd199f993ec6bdf902d3ae3cdce87710f4d546 Remove old junk files.
fe47737987532b1397cef91ef5160c24d3137e6c Update brcm_sai_version.h
07d928935f28476a4890e498c8541d907b33f6a3 Update README.md
0d35f6ddb3b5915b3cf1e6a75a32656eff22d956 SAI 4.2 EA1 release.
f9012cd44772d7ce4667d95bb95932f1720b4de6 Add missing build directories.
88399ae16a2c39996ee514f9c7d6064d77de8960 Temporary doc update.
0076ce9ccf0cea296fd9fa2fc0828e6da82bc7f0 Update README.md
ba56abdb8127569a47d219a15e2c519436509aa8 Cleanup unwanted file.
93fd029fa1d5c1d1025be129e83b53979ef4f617 SAI 4.2 Engg drop
1475c241e25c91fc8cb93d7c14cd70194bad1328 CSP10503275 request to pull KB0029314 into 3.7
```
**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by rlhui** [comment]
is it indeed that only one file needs changed? Thanks.

**Comment by rlhui** [comment]
"Build debain package with BCMSAI 4.3.1.3 and 6.5.19 hsdk." - typo?

**Comment by smaheshm** [comment]
rebuild vsimage please

**Comment by smaheshm** [comment]
retest vsimage please

**Comment by smaheshm** [comment]
> "Build debain package with BCMSAI 4.3.1.3 and 6.5.19 hsdk." - typo?

yes, it should be 4.2.1.3. Thanks.. will correct.

**Comment by smaheshm** [comment]
> is it indeed that only one file needs changed? Thanks.

Yes, the only change is updating the package name and the link.

**Comment by judyjoseph** [comment]
Check if we need a new GPL module for 6.5.19 SDK, we would need to update sonic-buildimage/platform/broadcom/saibcm-modules/ with the same.

File: platform/broadcom/saibcm-modules/make/Makefile.linux-x86-smp_generic_64-2_6 [hunk 1]
```diff
@@ -28,18 +28,29 @@ ifeq (,$(KFLAGS))
 KFLAGS := -nostdinc -isystem $(SYSINC) -I$(KERNDIR)/include -I$(KERNDIR)/arch/x86/include -include $(AUTOCONF) -D__KERNEL__ -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -Werror-implicit-function-declaration -Wno-format-security -fno-delete-null-pointer-checks -Os -m64 -mtune=generic -mno-red-zone -mcmodel=kernel -funit-at-a-time -maccumulate-outgoing-args -DCONFIG_AS_CFI=1 -DCONFIG_AS_CFI_SIGNAL_FRAME=1 -pipe -Wno-sign-compare -fno-asynchronous-unwind-tables -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -fno-stack-protector -fomit-frame-pointer -g -Wdeclaration-after-statement -Wno-pointer-sign
 endif
 
-ifeq ($(LINUX_MAKE_SHARED_LIB), 1)
-KFLAGS += -fPIC -mcmodel=small
-else
-KFLAGS += -fno-pie -mcmodel=kernel
```
Comment (prsunny) [inline platform/broadcom/saibcm-modules/make/Makefile.linux-x86-smp_generic_64-2_6:34]: `-fno-pie `was previously added for a compilation issue `gmodule.c:1:0: error: code model kernel does not support PIC mode`

File: platform/broadcom/saibcm-modules/debian/opennsl-modules.init [hunk 2]
```diff
@@ -53,17 +53,12 @@ function load_kernel_modules()
     modprobe linux-kernel-bde dmasize=$dmasize maxpayload=128 debug=4 dma_debug=1 usemsi=$usemsi
     modprobe linux-user-bde
 
-    # Using insmod with absolute path for psample to make sure bcm psample is loaded.
-    # There is a different psample.ko module getting created at net/psample/psample.ko 
-    insmod /lib/modules/$(uname -r)/extra/psample.ko
```
Comment (prsunny) [inline platform/broadcom/saibcm-modules/debian/opennsl-modules.init:58]: This is required for sflow support. We may want to ask BCM for providing this.

Comment (smaheshm) [reply platform/broadcom/saibcm-modules/debian/opennsl-modules.init:58]: These files are absent in 6.5.19 gpl modules. Will it work if psample related files are added back or should BRCM give us the psample code for 6.5.19.

Comment (prsunny) [reply platform/broadcom/saibcm-modules/debian/opennsl-modules.init:58]: It was added as part of 3.7 and with BUILD_PSAMPLE=1 (locally built driver as part of opennsl). We may want to get this clarified with Broadcom on what psample driver to be used.

**Comment by smaheshm** [comment]
rebuild broadcom please

**Comment by smaheshm** [comment]
retest broadcom please

**Comment by smaheshm** [comment]
retest broadcom please

**Comment by smaheshm** [comment]
retest broadcom please

**Comment by smaheshm** [comment]
Close in favor of #5532

**Pull request closed without merging**